java/src/main/java/ai/onnxruntime/ONNXAllocator.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright © 2019, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+
+/**
+ * An ONNX runtime memory allocator.
+ */
+public class ONNXAllocator implements AutoCloseable {
+
+    final long handle;
+
+    private boolean closed = false;
+
+    static {
+        try {
+            ONNX.init();
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load ONNX library",e);
+        }
+    }
+
+    /**
+     * Constructs the default CPU allocator.
+     * @throws ONNXException If the onnx runtime threw an error.
+     */
+    public ONNXAllocator() throws ONNXException {
+        handle = createAllocator(ONNX.ortApiHandle);
+    }
+
+    ONNXAllocator(long handle) {
+        this.handle = handle;
+    }
+
+    /**
+     * Create a Tensor from a Java primitive (or String) multidimensional array.
+     * The shape is inferred from the array using reflection.
+     * @param data The data to store in a tensor.
+     * @return An ONNXTensor storing the data.
+     * @throws ONNXException If the onnx runtime threw an error.
+     */
+    public ONNXTensor createTensor(Object data) throws ONNXException {
```

Comment:
I feel, these createXXX classes should be part of the Tensor class. Conceptully easier for the user.

Reply:
Should they accept an allocator as a parameter? They were previously on the session as that controlled all the memory.

Reply:
Moved onto OrtEnvironment as methods. They still need to reference an allocator, and the default allocator now lives inside there.